README.md

```diff
@@ -39,3 +39,27 @@ Some background:
 * There are as many gradient pairs as there are instances in a training dataset.
 * In order to find optimal splits for decision trees, we compute a **histogram** of gradients. Each bin of the histogram stands for a range of feature values. The value of the bin is given by the sum of gradients corresponding to the data points lying inside the range.
 * In each boosting iteration, we have to compute multiple histograms, each histogram corresponding to a set of instances.
+
+# Setting build types
+
+* By default, 'Release' build type will be used, with flags `-O3 -DNDEBUG`.
+
+* For perfiling, you may want to add debug symbols by choosing 'RelWithDebInfo' build type instead:
+  ```bash
+  cmake -DCMAKE_BUILD_TYPE=RelWithDebInfo ..
+  ```
+  This build type uses the following flags: `-O2 -g -DNDEBUG`.
+
+* For full control over the compilation flags, specify `CMAKE_CXX_FLAGS_RELEASE`:
+  ```bash
+  cmake -DCMAKE_CXX_FLAGS_RELEASE="-Ofast -DNDEBUG -march=native -mtune=native" ..
```

Comment:
`-march=native` is enough (no need `-mtune=native` when using `-march=native`, `march` defines the required target architecture to run on, while `mtune` defines a secondary targeted architecture to run on)

`-Ofast` not sure, it can scramble computations (better `-O3` for comparability with xgboost?). We need `-g` for profiling and mixing source code with computation time.

Reply:
It was just an example for customizing the flags.

Reply:
I changed it to more sensible flags